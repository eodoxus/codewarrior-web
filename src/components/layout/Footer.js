import React from "react";
import styles from "./Footer.scss";
import cx from "classnames";
import { AllHtmlEntities } from "html-entities";

export default ({ copy }) => {
  copy = new AllHtmlEntities().decode(copy);
  return <footer className={cx(styles.footer, "footer")}>{copy}</footer>;
};
