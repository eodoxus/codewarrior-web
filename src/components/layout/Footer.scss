@import "../../lib/lib.scss";

.footer {
  position: fixed;
  z-index: 9999;
  bottom: 0;
  left: 0;
  right: 0;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #ccc;
  border-top: 1px solid #999;
  color: $color-app-header;
}
