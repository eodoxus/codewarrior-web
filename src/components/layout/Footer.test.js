import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Footer from "./Footer";

jest.useFakeTimers();

describe("<Footer />", () => {
  let props;
  beforeEach(function() {
    props = {
      footer: "footy"
    };
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Footer copy={props.footer} />, div);
  });

  describe("snapshots", () => {
    it("collapses after a period of time when hide prop is set, then re-opens if hide prop changes to true", () => {
      const node = renderer.create(<Footer copy={props.footer} hide={true} />);
      expect(node.toJSON()).toMatchSnapshot("Open");
      jest.runTimersToTime(500);
      expect(node.toJSON()).toMatchSnapshot("Still open after 500ms");
      jest.runTimersToTime(100);
      expect(node.toJSON()).toMatchSnapshot("Collapsed after 600ms");

      node.update(<Footer copy={props.footer} hide={false} />);
      jest.runTimersToTime(500);
      expect(node.toJSON()).toMatchSnapshot("Still sclosed after 500ms");
      jest.runTimersToTime(100);
      expect(node.toJSON()).toMatchSnapshot("Open after 600ms");
      node.unmount();
    });
  });
});
