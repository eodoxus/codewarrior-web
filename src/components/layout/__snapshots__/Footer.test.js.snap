// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsed after 600ms 1`] = `
<footer
  className="collapsible collapsed footer"
>
  footy
</footer>
`;

exports[`Open 1`] = `
<footer
  className="collapsible footer"
>
  footy
</footer>
`;

exports[`Open after 600ms 1`] = `
<footer
  className="collapsible footer"
>
  footy
</footer>
`;

exports[`Still open after 500ms 1`] = `
<footer
  className="collapsible footer"
>
  footy
</footer>
`;

exports[`Still sclosed after 500ms 1`] = `
<footer
  className="collapsible collapsed footer"
>
  footy
</footer>
`;
